### PR TITLE
Lionelb/fix error modeles page

### PR DESCRIPTION
**Pull request by lionelB**
add `withError` Hoc to handle gracefully 4xx-5xx requests.
It allow to swap component rendering depending on the response.ok flag
In case of a fail request, it will display the _errorPage and let sentry
do its job.

### CODE REVIEW

File: packages/code-du-travail-frontend/pages/fiche-service-public.js [hunk 1]
```diff
@@ -21,7 +22,7 @@ const {
 } = getConfig();
 
 const fetchFiche = ({ slug }) =>
-  fetch(`${API_URL}/items/fiches_service_public/${slug}`).then(r => r.json());
+  fetch(`${API_URL}/items/fiches_service_public/${slug}`);
```
Comment (revolunet) [inline packages/code-du-travail-frontend/pages/fiche-service-public.js]: no more json ?

**Comment by revolunet** [comment]
looks great, but shouldn't this happen at `_app.js` level instead of every single page ?

**Comment by lionelB** [comment]
@revolunet it could, in that case I can remove the withErrorHoc and do the branching in app directly

**Comment by revolunet** [comment]
Awesome !

**Review by revolunet — APPROVED** [review]

File: packages/code-du-travail-frontend/pages/_error.js [hunk 2]
```diff
@@ -10,30 +10,33 @@ const {
   publicRuntimeConfig: { SENTRY_PUBLIC_DSN }
 } = getConfig();
 
-if (typeof window !== "undefined") {
+if (typeof window !== "undefined" && SENTRY_PUBLIC_DSN) {
   Sentry.init({ dsn: SENTRY_PUBLIC_DSN, debug: true });
```
Comment (UnbearableBear) [inline packages/code-du-travail-frontend/pages/_error.js:14]: C'est normal qu'on ait un sentry init deux fois (un dans _app et un dans _error) ?

Comment (lionelB) [reply packages/code-du-travail-frontend/pages/_error.js:14]: ca permet d'attraper les erreur dans app qui interviennent au moment du  render (componentDidCatch)

**Pull request merged**